### ISSUE TITLE
question about horovod on mxnet

### ISSUE THREAD

**Issue by JaggerYoung** [horovod/horovod#875]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) MXnet
2. Framework version: Newest
3. Horovod version: Newest
4. MPI version:  3.1.4
5. CUDA version: 8.0
6. NCCL version: 2.1.4
7. Python version: 2.7
8. OS and version: centos

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.md)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.md)?

**Your question:**
Please ask your question here.

I have successfully install horovod on mxnet. however, I get a result as followed which I
think is unreasonable. I use "mpirun -np 4 python mxnet_mnist.py --gpus=0,1" as example:

INFO:root:Epoch[0] Batch [50]	Speed: 3617.28 samples/sec	Train-accuracy=0.539828
INFO:root:Epoch[0] Batch [50]	Speed: 3630.34 samples/sec	Train-accuracy=0.529718
INFO:root:Epoch[0] Batch [100]	Speed: 8533.43 samples/sec	Train-accuracy=0.888750
INFO:root:Epoch[0] Batch [100]	Speed: 8506.38 samples/sec	Train-accuracy=0.882188
INFO:root:Epoch[0] Batch [150]	Speed: 5277.87 samples/sec	Train-accuracy=0.929375
INFO:root:Epoch[0] Batch [150]	Speed: 5264.60 samples/sec	Train-accuracy=0.920000
INFO:root:Epoch[0] Batch [200]	Speed: 6625.95 samples/sec	Train-accuracy=0.936875
INFO:root:Epoch[0] Batch [200]	Speed: 6402.29 samples/sec	Train-accuracy=0.945625
INFO:root:Epoch[0] Batch [250]	Speed: 5960.06 samples/sec	Train-accuracy=0.945312
INFO:root:Epoch[0] Batch [250]	Speed: 5791.01 samples/sec	Train-accuracy=0.951875
INFO:root:Epoch[0] Batch [300]	Speed: 6167.53 samples/sec	Train-accuracy=0.953750
INFO:root:Epoch[0] Batch [300]	Speed: 6165.04 samples/sec	Train-accuracy=0.966875
INFO:root:Epoch[0] Batch [350]	Speed: 8111.90 samples/sec	Train-accuracy=0.957500
INFO:root:Epoch[0] Batch [350]	Speed: 8151.89 samples/sec	Train-accuracy=0.960938
INFO:root:Epoch[0] Batch [400]	Speed: 7086.12 samples/sec	Train-accuracy=0.962812
INFO:root:Epoch[0] Batch [400]	Speed: 6877.42 samples/sec	Train-accuracy=0.961250
INFO:root:Epoch[0] Batch [450]	Speed: 6896.63 samples/sec	Train-accuracy=0.962187
INFO:root:Epoch[0] Batch [450]	Speed: 6686.42 samples/sec	Train-accuracy=0.967187

Why the same iteration get differenct acc. Sync batch does not work？
Have I made some mistake？ I wish for your suggestion.

**Comment by eric-haibin-lin** [comment]
It looks like the metric is just computed per process, not reduced between nodes. 

@apeforest @yuxihu
